Source/WebCore/dom/Document.h

```diff
@@ -2419,6 +2424,8 @@ class Document
 
     bool m_scheduledDeferredAXObjectCacheUpdate { false };
 
+    bool m_isInRenderingUpdate { false };
```

Comment:
I'm not sure this is the best approach. This new state on Document replicates some of what `m_renderingUpdateRemainingSteps` represents on Page.

Reply:
Good point, indeed m_isInRenderingUpdate is not needed.